### PR TITLE
Add Scaleway ip address management

### PR DESCRIPTION
**Pull request by abarbare**
##### SUMMARY

This merge request adds a Scaleway module to manage IPv4 address on a Scaleway instance. It is especially useful to use a non-dynamic IP address to be able to move this IP to another server.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

scaleway_ip

##### ANSIBLE VERSION

```
ansible 2.6.0
  config file = /Users/abarbare/.ansible.cfg
  configured module search path = [u'/Users/abarbare/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.6.0/libexec/lib/python2.7/site-packages/ansible
  executable location = /usr/local/Cellar/ansible/2.6.0/bin/ansible
  python version = 2.7.15 (default, Jun 17 2018, 12:46:58) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
```

### CODE REVIEW

**Comment by abarbare** [comment]
cc @sieben

File: test/legacy/scaleway_ip_address.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# SCW_API_KEY='XXX' ansible-playbook ./test/legacy/scaleway_user_data.yml
```
Comment (remyleone) [inline test/legacy/scaleway_ip_address.yml]: Update with the correct file path

**Review by remyleone — APPROVED** [review]

File: test/legacy/scaleway_ip_address.yml (hunk 1 above)
Comment (abarbare) [reply test/legacy/scaleway_ip_address.yml]: Well catch 👍

**Comment by abarbare** [comment]
It seems to be all green here 😃 
Are you waiting something more from my side?

**Comment by remyleone** [comment]
All good for me. @maxamillion Could you merge?

File: lib/ansible/modules/cloud/scaleway/scaleway_ip.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+                        'SCW_TOKEN', 'SCW_API_KEY', 'SCW_OAUTH_TOKEN']),
+                required=True,
+            ),
+            state=dict(choices=['present', 'absent'], required=True),
+            organization=dict(required=True),
+            server_id=dict(required=True),
+            timeout=dict(type='int', default=30),
+            region=dict(required=True, choices=SCALEWAY_LOCATION.keys()),
+        ),
+        supports_check_mode=True,
```
Comment (maxamillion) [inline lib/ansible/modules/cloud/scaleway/scaleway_ip.py:214]: It seems as though the code isn't checking for `check_mode` and exiting before a change occurs, am I missing something?

Comment (abarbare) [reply lib/ansible/modules/cloud/scaleway/scaleway_ip.py:214]: Indeed, my mistake! 
Thanks for your feedback 👍

**Comment by abarbare** [comment]
Let me know if you want me to squash my commits

**Comment by remyleone** [comment]
Yup I think a squash is a good idea.

**Comment by abarbare** [comment]
squashed

**Comment by remyleone** [comment]
@Spredzy Do you think it should be rebase on top of https://github.com/ansible/ansible/pull/44377/commits/c229e2a956ef3700177671a979effe00005f565e ?

**Comment by remyleone** [comment]
@abarbare How can you attach an IP to a compute instance?

**Comment by Spredzy** [comment]
@abarbare thanks for the review. We've @sieben we've been working on refactoring the existing modules, could you rebase your patch on top of latest patchset in https://github.com/ansible/ansible/pull/44377 and apply the same pattern applied for the other modules already present (ie. `scaleway_compute` or `scaleway_volume`)

Let me know if you need any guidance

**Comment by Spredzy** [comment]
@abarbare https://github.com/ansible/ansible/pull/44377 has been merged. So you simply have to rebase it against `devel`.

**Comment by abarbare** [comment]
Ack, I'll rebase and apply the refactoring you did on others modules 👍

**Comment by remyleone** [comment]
Superseded by https://github.com/ansible/ansible/pull/45121

**Pull request closed without merging**